### DISCUSSION TITLE
quarkus:dependency-tree failure due to Failed to inject extension deployment dependencies for quarkus-smallrye-reactive-messaging-amqp-deployment.

### DISCUSSION THREAD

**Discussion by whitingjr** [quarkusio/quarkus#41748] [Q&A]
Hi,
 I am seeing an error when using the quarkus plugin to show the dependency tree for this [pom.xml](https://gist.github.com/whitingjr/5aa95b18dff7f9805b20423c74ae4cd1).

```
[INFO] Scanning for projects...
[INFO] 
[INFO] ---------------------< io.jenkins.plugins:horreum >---------------------
[INFO] Building Horreum Plugin 0.1.5-SNAPSHOT
[INFO]   from pom.xml
[INFO] --------------------------------[ hpi ]---------------------------------
[INFO] 
[INFO] --- quarkus:3.10.2:dependency-tree (default-cli) @ horreum ---
[INFO] Quarkus application PROD mode build dependency tree:
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.801 s
[INFO] Finished at: 2024-07-08T09:21:01+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:3.10.2:dependency-tree (default-cli) on project horreum: Failed to resolve application model io.jenkins.plugins:horreum::pom:0.1.5-SNAPSHOT dependencies: Failed to inject extension deployment dependencies for io.jenkins.plugins:horreum:pom:0.1.5-SNAPSHOT: Quarkus extension deployment artifact io.quarkus:quarkus-smallrye-reactive-messaging-amqp-deployment:jar:3.8.4 does not appear to depend on the corresponding runtime artifact io.hyperfoil.tools:horreum-integration-tests:jar:0.13.1 -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

 I have attempted adding to the project the Quarkus extension `quarkus-smallrye-reactive-messaging-amqp` and the `quarkus-smallrye-reactive-messaging-amqp-deployment:3.8.4`. But that did not solve this issue.

 Any recommendations on how to solve this ?

[a comment by an automated account is not shown]

**Comment by cescoffier** [comment]
Can you provide a reproducer? It looks very weird. I believe there is an issue in an extension metadata.

**Reply by whitingjr** [reply]
@cescoffier to provide a simplified reproducer is going to be too time consuming due to deadlines. The next best I can offer is this link to the [commit](https://github.com/whitingjr/horreum-plugin/commit/8d2aa11ff97d1c1681dc43bd8c4f3c0cee52e41f) on the project that produces the message.
btw, I am working around this by splitting up the module to 2 sub-modules. Which might mean this error seen is moot to me but possibly useful for the project as a crazy corner case. I recognised the quarkus-maven-plugin depends on the module packaging type to be "jar". But this module uses "hpi" packaging because it is a jenkins plugin.

**Reply by cescoffier** [reply]
Well, the commit seems to be the switch from Quarkus 2 to Quarkus 3. It's pretty large.

I looked at the pom:

1. you use Java 11; we strongly recommend updating. 
2. Some dependencies are managed by the Quarkus BOM (smallrye commons, JBoss logging and so on), do **NOT** override these versions


I will try to checkout the project and have a look, but I cannot commit on a date.

**Reply by cescoffier** [reply]
`io.hyperfoil.tools:horreum-integration-tests:jar:0.13.1` is particularly odd, it contains a LOT of stuff, which it should not (some part of vertx, some parts of reactive messaging...). I would look into this one.

**Reply by cescoffier** [reply]
Typically, the horreum-integration-tests jar contains a `quarkus-extension.properties` with: `deployment-artifact=io.quarkus\:quarkus-smallrye-reactive-messaging-amqp-deployment\:3.8.4`. So, I would not depend on this jar or would do it properly.

**Reply by whitingjr** [reply]
@cescoffier updating the dependency configuration for horreum-integration-tests to the **test-jar** type avoids the shaded classes. This issue is solved. ty
